datafusion/common/src/config.rs

```diff
@@ -1874,4 +2042,193 @@ mod tests {
+            let configured_col_props = extract_column_options(props, configured_col);
+
+            let key_value_metadata = props
+                .key_value_metadata()
+                .map(|pairs| {
+                    HashMap::from_iter(
+                        pairs
+                            .iter()
+                            .cloned()
+                            .map(|KeyValue { key, value }| (key, value)),
+                    )
+                })
+                .unwrap_or_default();
+
+            TableParquetOptions {
+                global: ParquetOptions {
+                    // global options
+                    data_pagesize_limit: props.dictionary_page_size_limit(),
+                    write_batch_size: props.write_batch_size(),
+                    writer_version: format!("{}.0", props.writer_version().as_num()),
+                    dictionary_page_size_limit: props.dictionary_page_size_limit(),
+                    max_row_group_size: props.max_row_group_size(),
+                    created_by: props.created_by().to_string(),
+                    column_index_truncate_length: props.column_index_truncate_length(),
+                    data_page_row_count_limit: props.data_page_row_count_limit(),
+
+                    // global options which set the default column props
+                    encoding: default_col_props.encoding,
+                    compression: default_col_props.compression,
+                    dictionary_enabled: default_col_props.dictionary_enabled,
+                    statistics_enabled: default_col_props.statistics_enabled,
+                    max_statistics_size: default_col_props.max_statistics_size,
+                    bloom_filter_on_write: default_col_props
+                        .bloom_filter_enabled
+                        .unwrap_or_default(),
+                    bloom_filter_fpp: default_col_props.bloom_filter_fpp,
+                    bloom_filter_ndv: default_col_props.bloom_filter_ndv,
+
+                    // not in WriterProperties
+                    enable_page_index: Default::default(),
+                    pruning: Default::default(),
+                    skip_metadata: Default::default(),
+                    metadata_size_hint: Default::default(),
+                    pushdown_filters: Default::default(),
+                    reorder_filters: Default::default(),
+                    allow_single_file_parallelism: Default::default(),
+                    maximum_parallel_row_group_writers: Default::default(),
+                    maximum_buffered_record_batches_per_stream: Default::default(),
+                    bloom_filter_on_read: Default::default(),
+                },
+                column_specific_options: HashMap::from([(
+                    COL_NAME.into(),
+                    configured_col_props,
+                )]),
+                key_value_metadata,
+            }
+        }
+
+        #[test]
```

Comment:
Maybe we could also add a test here that described in https://github.com/apache/datafusion/issues/11367 -- that takes a default `ParquetOptions`, creates the `ArrowWriterOption` and then ensures the fields are the same as `ArrowWriterOptions::default()`

Reply:
Added the test in [this commit](https://github.com/apache/datafusion/pull/11444/commits/a484974239274b2423110bdf4f6fda0478e712c2), but did the equivalency assertion a bit differently. Parquet's `WriterProperties` and it's builder do not implement Eq or PartialEq. 

Rather than asserting each prop independently based upon the getter, I elected to (a) explicitly define the props that do not match, and then (b) use the same identify function (`session_config_from_writer_props`) for the assertion.

Reply:
I had to make [a slight change](https://github.com/apache/datafusion/pull/11444/commits/ad9f695b69336fa7e1cab8cba7e59587428c2a2e) to avoid a test regression in the sqllogictests.

That^^ change (to avoid test regression) also brings us further from the matching-default-behaviors between the extern parquet vs datafusion. I [added more tests](https://github.com/apache/datafusion/pull/11444/commits/ab950c56142084384df466fa949ee7a1eac70f6d) to demonstrate this divergence.

Reply:
In the follow up PR, I plan to propose (and open for discussion) which defaults should be changed in datafusion vs parquet.

Reply:
Per request, moving these into a follow up PR. (Make more reviewable.)